azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/VirtualMachineScaleSet.java

```diff
@@ -1739,8 +1739,25 @@ WithManagedCreate withNewDataDiskFromImage(int imageLun,
              *
              * @return the next stage of the definition
              */
-            @Beta(Beta.SinceVersion.V1_11_0)
+            @Deprecated
             WithCreate withLowPriorityVirtualMachine(VirtualMachineEvictionPolicyTypes policy);
+
+            /**
+             * Specify that virtual machines in the scale set should be spot priority VMs.
+             *
+             * @return the next stage of the definition
+             */
+            WithCreate withSpotPriorityVirtualMachine();
```

Comment:
Probably just `withSpotVirtualMachine`, as in doc?

Reply:
If so, should we set `withSpot()` instead of `withSpotPriority()` in VM?

Reply:
OK, `withSpot` seems not very intuitive as well. Then keep it as `withSpotPriorityVirtualMachine`.

Keep the PR open for a while to see if there is other opinions.